### PR TITLE
 Enable AVX512 Additional 16 SIMD Registers, with accessors instead of macros

### PR DESCRIPTION
**Pull request by BruceForstall**
This is https://github.com/dotnet/runtime/pull/79544 with one additional change.

### CODE REVIEW

File: src/coreclr/jit/codegen.h [hunk 1]
```diff
@@ -35,6 +35,17 @@ class CodeGen final : public CodeGenInterface
         GenTree* addr, bool fold, bool* revPtr, GenTree** rv1Ptr, GenTree** rv2Ptr, unsigned* mulPtr, ssize_t* cnsPtr);
 
 private:
+#if defined(TARGET_AMD64)
+    regMaskTP get_RBM_ALLFLOAT() const
+    {
+        return compiler->rbmAllFloat;
```
Comment (kunalspathak) [inline src/coreclr/jit/codegen.h:41]: Why not `compiler->get_RBM_ALLFLOAT()` and likewise for other methods in individual classes?

**Comment by kunalspathak** [comment]
Approach seems reasonable and hoping it won't affect the TP.

File: src/coreclr/jit/codegen.h (hunk 1 above)
Comment (BruceForstall) [reply src/coreclr/jit/codegen.h:41]: Either way is fine; they're equivalent. Your suggestion is slightly better encapsulation. However, I'd like to do an experiment to see how much TP could be gained by making a local copy of these fields -- probably a minimal effect.

**Comment by BruceForstall** [comment]
All failures look known or infra

**Comment by BruceForstall** [comment]
This change was merged into #79544

**Pull request closed without merging**